pkg/proc/variables.go

```diff
@@ -1724,6 +1749,100 @@ func (v *Variable) loadInterface(recurseLevel int, loadData bool, cfg LoadConfig
 	}
 }
 
+// ConstDescr describes the value of v using constants.
+func (v *Variable) ConstDescr() string {
+	if v.bi == nil || (v.Flags&VariableConstant != 0) {
+		return ""
+	}
+	ctyp := v.bi.consts.Get(v.DwarfType)
+	if ctyp == nil {
+		return ""
+	}
+	if typename := v.DwarfType.Common().Name; strings.Index(typename, ".") < 0 || strings.HasPrefix(typename, "C.") {
+		// only attempt to use constants for user defined type, otherwise every
```

Comment:
I don't understand this comment.

This patch seems to index constants by type, and then use the value & type information to evaluate the constant. I'm curious why we don't simply treat constants like other variables and evaluate them by name to avoid this confusion.

Reply:
Is this to support nicer describes by being able to represent constants as bitwise OR and such?

Reply:
Yes. Take for example the type `proc.FloatSpecial`, I think when we print variables of types like that (that are essentially enums) the name of the constant should be printed instead.